modules/standard/FFTW.chpl

```diff
@@ -117,23 +118,67 @@ module FFTW {
   // more native Chapel types anyway.
 
   /*
-    Creates a plan for a complex->complex DFT.
+    Creates a plan for an out-of-place complex-to-complex DFT.
 
     :arg input: The input array, which can be of any rank
     :type input: [] `complex(128)`
 
-    :arg output: The output array, with rank matching the input array's
+    :arg output: The output array, whose size and shape must match the input array's
     :type output: [] `complex(128)`
     
-    :arg sign: The sign of the exponent in the DFT formula.
+    :arg sign: :const:`FFTW_FORWARD` or :const:`FFTW_BACKWARD`
     :type sign: c_int
 
-    :arg flags: planning-rigor and/or algorithm-restriction flags
+    :arg flags: the bitwise-or of any planning-rigor or algorithm-restriction flags that should be used in creating the plan (e.g., :const:`FFTW_MEASURE` ``|`` :const:`FFTW_PRESERVE_INPUT`)
+    :type flags: c_int
+
+    :returns: The :type:`fftw_plan` representing the resulting plan
+  */
+  proc plan_dft(input: [?Din] complex(128), output: [?Dout] complex(128), 
+                 sign: c_int, flags: c_uint) : fftw_plan
+  {
+    if !noFFTWsizeChecks {
+      var error = false;
+
+      for i in 1..input.rank do
+        error |= Private_FFTW.checkDimMatch(Din, Dout, i, "plan_dft()");
+
+      if error then
+        halt("Incorrect array sizes in plan_dft()");
+    }
+
+    return plan_dft_help(input, output, sign, flags);
+  }
+
+
+  /*
+    Creates a plan for an in-place complex-to-complex DFT.
+
+    :arg arr: The array to use as the in-place input/output array.
+    :type arr: [] `complex(128)`
+
+    :arg sign: :const:`FFTW_FORWARD` or :const:`FFTW_BACKWARD`
     :type sign: c_int
 
-    :returns: The fftw_plan representing the plan
+    :arg flags: the bitwise-or of any planning-rigor or algorithm-restriction flags that should be used in creating the plan (e.g., :const:`FFTW_MEASURE` ``|`` :const:`FFTW_PRESERVE_INPUT`)
+    :type flags: c_int
+
+    :returns: The :type:`fftw_plan` representing the resulting plan
   */
-  proc plan_dft(input: [] complex(128), output: [] complex(128), 
+  proc plan_dft(arr: [] complex(128), sign: c_int, flags: c_uint): fftw_plan {
+    return plan_dft_help(arr, arr, sign, flags);
+  }
+
+  //
```

Comment:
I am interested that you've taken the path of defining an internal module Private_FFTW for checkDimMatch.  Longer
term it might be appropriate to move plan_dft_help() in there too.